pylint/checkers/base.py

```diff
@@ -605,8 +606,13 @@ def _check_redefinition(self, redeftype, node):
         """check for redefinition of a function / method / class name"""
         defined_self = node.parent.frame()[node.name]
         if defined_self is not node and not astroid.are_exclusive(node, defined_self):
-            self.add_message('function-redefined', node=node,
-                             args=(redeftype, defined_self.fromlineno))
+            # before emitting a message check that function/method name doesn't not
+            # correspond to dummy names.
+            dummy_variables_rgx = lint_utils.get_global_option(
+                self, 'dummy-variables-rgx', default=None)
+            if dummy_variables_rgx.match(defined_self.name) is None:
```

Comment:
what if `dummy_variables_rgx` is None?
Also, we can simplify `... is None` check to `if not ...`